core/src/main/java/org/apache/iceberg/ManifestEntry.java

```diff
@@ -49,7 +50,7 @@ public int id() {
 
   private final org.apache.avro.Schema schema;
   private Status status = Status.EXISTING;
-  private long snapshotId = 0L;
+  private Long snapshotId = null;
```

Comment:
Just to make sure: have you checked that older readers can read files produced with optional instead of required snapshotIds?

Reply:
Yes, I tested this locally.